pkg/operator/injection/injection.go

```diff
@@ -63,6 +65,19 @@ func WithOptionsOrDie(ctx context.Context, opts ...options.Injectable) context.C
 	return ctx
 }
 
+func WithClient(ctx context.Context, client client.Client) context.Context {
+	return context.WithValue(ctx, clientKey, client)
+}
+
+func GetClient(ctx context.Context) client.Client {
```

Comment:
IIRC @jonathan-innis intentionally was avoiding this pattern, though I don't know his rationale. I don't see a great alternative though.

Reply:
I don't personally see a way around it either